app/services/api_bridge/query.rb

```diff
@@ -135,10 +136,14 @@ def query(request_options={})
 
     # rows is a deprecated option, please use num instead
     # still supporting for backwards compatibility
+    # overriding num with rows because if an app is specifically
+    # sending a rows request, then we need to override the default
+    # num set when class was initialized
     def remove_rows(opts={})
       if opts.key?("rows")
-        opts["num"] = opts["rows"] if !opts.key?("num")
+        opts["num"] = opts["rows"]
         opts.delete("rows")
+        Rails.logger.warn("Parameter 'rows' is deprecated, please use 'num' instead")
```

Comment:
Perhaps add a TODO comment note above this to remove in a future version.

Or if we start needing to deprecate more things we could use a custom `DEPRECATE` note :man_shrugging: : https://guides.rubyonrails.org/command_line.html#notes